Lombiq.JsonEditor/Controllers/AdminController.cs

```diff
@@ -0,0 +1,105 @@
+﻿using Lombiq.HelpfulLibraries.OrchardCore.Contents;
+using Lombiq.JsonEditor.ViewModels;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Localization;
+using Newtonsoft.Json;
+using OrchardCore.ContentManagement;
+using OrchardCore.Contents;
+using OrchardCore.DisplayManagement;
+using OrchardCore.DisplayManagement.Layout;
+using OrchardCore.Title.ViewModels;
+using System.Threading.Tasks;
+using YesSql;
+
+namespace Lombiq.JsonEditor.Controllers;
+
+public class AdminController : Controller
+{
+    private readonly IAuthorizationService _authorizationService;
+    private readonly IContentManager _contentManager;
+    private readonly ILayoutAccessor _layoutAccessor;
+    private readonly ISession _session;
+    private readonly IShapeFactory _shapeFactory;
+    private readonly IStringLocalizer<AdminController> T;
+
+    public AdminController(
+        IAuthorizationService authorizationService,
+        IContentManager contentManager,
+        ILayoutAccessor layoutAccessor,
+        ISession session,
+        IShapeFactory shapeFactory,
+        IStringLocalizer<AdminController> stringLocalizer)
+    {
+        _authorizationService = authorizationService;
+        _contentManager = contentManager;
+        _layoutAccessor = layoutAccessor;
+        _session = session;
+        _shapeFactory = shapeFactory;
+        T = stringLocalizer;
+    }
+
+    public async Task<IActionResult> Edit(string contentItemId)
+    {
+        if (string.IsNullOrWhiteSpace(contentItemId) ||
+            await _contentManager.GetAsync(contentItemId, VersionOptions.Latest) is not { } contentItem ||
+            !await CanEditAsync(contentItem))
+        {
+            return NotFound();
+        }
+
+        var titleShape = await _shapeFactory.CreateAsync<TitlePartViewModel>("TitlePart", model =>
+        {
+            model.Title = T["Edit {0} as JSON", contentItem.ContentType];
```

Comment:
```suggestion
            model.Title = T["Edit \"{0}\" as JSON", contentItem.DisplayText];
```

Reply:
Having the `DisplayText` displayed somewhere would still be nice, because you can have a lot of identically titled tabs otherwise.